src/Features/CSharp/Portable/ExtractMethod/CSharpMethodExtractor.CSharpCodeGenerator.ExpressionCodeGenerator.cs

```diff
@@ -189,12 +189,11 @@ protected override SyntaxNode GetFirstStatementOrInitializerSelectedAtCallSite()
                 protected override SyntaxNode GetLastStatementOrInitializerSelectedAtCallSite()
                     => GetFirstStatementOrInitializerSelectedAtCallSite();
 
-                protected override async Task<SyntaxNode> GetStatementOrInitializerContainingInvocationToExtractedMethodAsync(
-                    SyntaxAnnotation callSiteAnnotation, CancellationToken cancellationToken)
+                protected override async Task<SyntaxNode> GetStatementOrInitializerContainingInvocationToExtractedMethodAsync(CancellationToken cancellationToken)
                 {
                     var enclosingStatement = GetFirstStatementOrInitializerSelectedAtCallSite();
-                    var callSignature = CreateCallSignature().WithAdditionalAnnotations(callSiteAnnotation);
-                    var invocation = callSignature.IsKind(SyntaxKind.AwaitExpression, out AwaitExpressionSyntax awaitExpr) ? awaitExpr.Expression : callSignature;
+
+                    var (invocation, invocationWithAwaitOpt) = CreateCallSignatureParts();
```

Comment:
it took me a ton of time to realize what was going on.  CreateCallSignature may return an `await Foo()` or `Foo()` call.  but then later code tries to deconstruct that so it can just touchup the `Foo()` portion regardless of what we got back.  This cahnge actually has us return both parts to make this strongly typed (i.e. we get a true InvocationExpression node back) and make it clearer why certain things are safe.

Note: this is undone in a couple of commits as there's a much cleaner way to do this altogether.